openquake/hazardlib/geo/surface/base.py

```diff
@@ -427,6 +427,40 @@ def get_middle_point(self):
 
         return mesh.get_middle_point()
 
+    def get_resampled_top_edge(self, angle_var=0.1):
+        """
+        Compute the top edge of the rupture plane.
+
+        :param angle_var:
+            Float number represents the angel allowed to bend within a fault
+            segment.
+        :returns:
+            A :class:`~openquake.hazardlib.geo.line.Line` representing the
+            rupture surface's top edge.
+        """
+        mesh = self.get_mesh()
+        top_edge = []
+        top_edge.append(Point(mesh.lons[0][0], mesh.lats[0][0],
+                              mesh.depths[0][0]))
+        for i in range(len(mesh.triangulate()[1][0]) - 1):
```

Comment:
Wouldn't be more efficient to create an object corresponding to the triangulated mesh and then work with it so as to avoid to continuously triangulating the mesh?